_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,265 @@
+    uses a channel he has with Mallory to send her a related HTLC:
+
+     - If Mallory discloses the preimage for `<hash>`, she can spend 1
+       BTC of Bob's money (we're ignoring routing fees in this example)
+
+     - Otherwise, after 40 blocks, Bob can spend his 1 BTC back to
+       himself
+
+    Although the above HTLCs are usually created and settled offchain,
+    each party also has a *commitment transaction* they can use to
+    put the HTLC commitment onchain.  A separate onchain *settlement
+    transaction* can fulfill either condition of the HTLC.
+
+    For example, Mallory can publish the commitment transaction and then
+    create a settlement transaction that provides the preimage and
+    claims Bob's 1 BTC.  If Bob sees Mallory's preimage settlement
+    transaction before the 80 block timeout from the Alice-Bob contract,
+    Bob can extract that preimage and use it to claim the 1 BTC from
+    Alice (either onchain or offchain).  Or, if Bob doesn't see a
+    preimage settlement transaction, Bob can create his own refund
+    settlement transaction after 40 blocks that takes back his 1 BTC,
+    allowing him to also refund Alice her 1 BTC (again, either onchain
+    or offchain).  In either case, this leaves everyone in compliance
+    with the intent of their contracts.
+
+    Unfortunately, as disclosed this week, there appears to be a way for
+    Mallory to circumvent the process by both preventing Bob from
+    learning the preimage while also preventing him from sending his
+    refund settlement transaction.
+
+    - **Preimage denial:** Mallory can prevent Bob from learning the
+      preimage by giving her preimage settlement transaction a low
+      feerate that keeps it from being confirmed quickly.  If Bob is
+      only looking for preimages in the block chain, he won't see
+      Mallory's transaction while it remains unconfirmed.
+
+    - **Refund denial:** Mallory's prior broadcast of the preimage
+      settlement transaction can prevent miners and Bitcoin relay nodes
+      from accepting Bob's later broadcast of the refund settlement
+      transaction because the two transactions *conflict*, meaning they
+      both spend the same input (a UTXO created in the commitment
+      transaction).  In theory, Bob's refund settlement transaction will
+      pay a higher feerate and so can [replace][topic rbf] Mallory's
+      preimage settlement but, in practice, Mallory can use various
+      [transaction pinning][topic transaction pinning] techniques to
+      prevent that replacement from happening.
+
+    Because Bob is prevented from either learning about the preimage
+    settlement transaction or getting his refund settlement transaction
+    confirmed, Alice is able to reclaim the 1 BTC she offered Bob in the
+    Alice-Bob HTLC once its 80 block timeout expires.  When Mallory's
+    preimage settlement transaction does eventually confirm, Mallory
```

Comment:
Mallory can arrange directly with a miner, paying fees directly instead of in-transaction, e.g. https://www.viabtc.com/tools/txaccelerator/ , which prevents the fee bump from being seen in *any* mempool.

Reply:
True.  However, I don't think it's strictly necessary for Mallory to fee bump at all.  As long as she can keep her transaction in miner mempools and that transaction is not being replaced, Mallory's transaction is likely to eventually confirm and so give her the money.  Since it's not required for Mallory to fee bump, I've left it out of the description (that said, Mallory probably would want to use something like the viabtc thing to lock in her gains).